innovation_sweet_spots/pipeline/pilot/investment_predictions/link_cb_gtr/link_cb_gtr.py

```diff
@@ -0,0 +1,130 @@
+from innovation_sweet_spots import PROJECT_DIR
+from jacc_hammer.name_clean import preproc_names
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from jacc_hammer.fuzzy_hash import Cos_config, Fuzzy_config, match_names_stream
+import pandas as pd
+from utils import merge_matches_with_cb_and_gtr
+from typing import List
+import typer
+
+
+STOPWORDS = ["ltd", "llp", "limited", "holdings", "group", "cic", "uk", "plc"]
+CB_PATH = PROJECT_DIR / "inputs/data/cb/crunchbase_organizations.csv"
+GTR_PATH = PROJECT_DIR / "inputs/data/gtr/gtr_organisations.csv"
+CB_GTR_LINK_SAVE_PATH = PROJECT_DIR / "inputs/data/cb_gtr_link"
+
+
+def link_cb_to_gtr(
+    chunksize: int = 100_000,
+    sim_mean_min: int = 92,
+    stopwords: List[str] = STOPWORDS,
+):
+    """Link crunchbase organisation ids to gateway to research organisation ids
+    using fuzzy matching on the organisation names.
```

Comment:
Hi @Jack-0-0 , thank you for this - a very important contribution!

I think it would be helpful to explain in slightly greater detail how the matching works - perhaps in the docstring?

For example:
- Is it one-to-one or one-to-many matching? (is one Crunchbase organisation matched only to one GtR organisation or several?)
- Do mention that you're using both `name` and `legal_name`... and if there are two different matches for one organisations, do you keep both?
- How does the threshold `sim_mean_min` work: Do you automatically reject all matches below this, or is it still possible after running this script to manually review the matches that are below this threshold?

Also: Is it easy to flip the matching, and try finding for each GtR organisation a corresponding Crunchbase organisation?

Reply:
> Hi @Jack-0-0 , thank you for this - a very important contribution!
> 
> I think it would be helpful to explain in a slightly greater detail how the matching works - perhaps in the docstring?
> 
> For example:
> 
> * Is it one-to-one or one-to-many matching? (is one Crunchbase organisation matched only to one GtR organisation or several?)
> * Do mention that you're using both `name` and `legal_name`... and if there are two different matches for one organisations, do you keep both?
> * How does the threshold `sim_mean_min` work: Do you automatically reject all matches below this, or is it still possible, after running this script, to manually review the matches that are below this threshold?
> 
> Also: Is it easy to flip the matching, and try finding for each GtR organisation a corresponding Crunchbase organisation?

Thanks @beingkk.
* The crunchbase organisation is matched to several GtR organisations
* If there is a different GtR match for a crunchbase organisation on `name` and `legal_name` then both GtR matches are kept.
* `sim_mean_min` -- all matches below this are filtered out. If you wanted to review the matches I would set the `sim_mean_min` lower and then look at the `cb_gtr_id_names_and_addresses.csv` file.

I have added information related to the above to the docstring.

It is quite easy to flip the matching. The `link_cb_to_gtr` function is not setup to do that currently but it wouldn't take too much work to change it. The matching is done by the function ` match_names_stream` and so here the variables would need to be swapped around.

Reply:
> Also: What is the recommended next step after running this script? (eg, manual inspection?)

I started with a lower `sim_mean_min` and inspected the results in `cb_gtr_id_names_and_addresses.csv`. I thought matches with a `sim_mean_min` of 92 looked like good matches and was a good starting point.

There are also some good matches at `sim_mean_min` of 75+ that we could manually review and pick out at a later date.